src/server/svr_func.c

```diff
@@ -3897,6 +3986,64 @@ set_entity_resc_sum_max(job *pjob, pbs_queue *pque, attribute *altered_resc,
 		return rc_final;
 	return 0;
 }
+/**
+ * @brief
+ * 		account_entity_limit_usages() - set entity usage
+ *		for all four combination of entity limits res/ct and max/queued
+ *		1. Called against server attributes (pque will be null):
+ *	   		a. When new job arrives (INCR)
+ *	   		b. when job is purged (DECR)
+ *		2. Called against queue attributes (pque will point to queue struct):
+ *	   		a. on any enqueue (INCR), or
+ *	   		b. on any dequeue (DECR)
+ *
+ * @param[in]	pjob	-	pointer to job structure
+ * @param[in]	pque	-	pque will point to queue structure, i.e. not be null
+ * @param[in]	altered_resc	-	altered resources.
+ * @param[in]	op	-	operator example- INCR, DECR
+ *
+ * @return	int
+ * @retval	zero	: all went ok
+ * @retval	PBS_Enumber	: if error, typically a system or internal error
+ */
+int
+account_entity_limit_usages(job *pjob, pbs_queue *pque, attribute *altered_resc,
+	enum batch_op op)
+{
+	int rc,ret_error = PBSE_NONE;
+
+	/* not doing NULL checks of parameters as this function is currently invoked from sane locations */
+
+	if ((rc=set_entity_ct_sum_max(pjob, pque, op)) != 0) {
```

Comment:
If you are hitting an error, why don't you return from the function?

Reply:
For the answer you need to see the fixed bug [PP-406](https://pbspro.atlassian.net/browse/PP-406) *Error reporting code path fails to report all of the errors when setting entity count*

Reply:
Thanks for the explanation. I was just making sure that these changes are not accidental.